lib/sync-srv.js

```diff
@@ -156,47 +156,44 @@ function doClientSync(dataset_id, params, callback) {
   function clientSyncImpl() {
 
     DataSetModel.getDatasetClient(dataset_id, params, function (err, datasetClient) {
-      if (err) return callback(err);
 
-      //Deal with any Acknowledgement of updates from the client
+      if (err) {
+        return callback(err);
+      }
+
+      if (datasetClient.dataHash) {
+        var hashMatch = (datasetClient.dataHash === params.dataset_hash);
+        
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the useless trailing whitespaces at the end of this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)